api/auth/access-token-provider.ts

```diff
@@ -0,0 +1,39 @@
+import { SmartlingAuthApi } from "./index";
+import { AccessTokenDto } from "./dto/access-token-dto";
+
+export class AccessTokenProvider extends SmartlingAuthApi {
```

Comment:
I'm not happy that we inherited from a large class that contains code that is useless to us. So we go against OOP principles.
Instead, we only need to define and implement an interface with two methods (get access token, get token type). All SDK classes should have access to the actual implementation by referencing the interface methods.

Reply:
> So we go against OOP principles.
technically we don't. everywhere where auth api client is expected we can pass access token provider with no issues. it literally follow LSP (Liskov Substitution Principle) which doesn't state we only need type with interface for it to work, it might be a root class with no problems.

I thought about it but didn't want to do that because we only have 2 auth clients implementations. there will never be more. but now I'm convinced. Ok, will rework.